File: app/src/main/java/com/weiminal/biliwidget/DataToJson.java
package com.weiminal.biliwidget;

import org.json.JSONException;
import org.json.JSONObject;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.Reader;
import java.net.URL;
import java.nio.charset.Charset;

public class DataToJson {

    /** 数据获取 1.0
     *
     * by WeiMinal **/

    public DataToJson(){

    }

    private static String str(Reader r) throws IOException {
        StringBuilder sb = new StringBuilder();
        int a;
        while ((a = r.read()) != -1) {
            sb.append((char) a);
        }
        return sb.toString();
    }

    public static JSONObject getData(String url) throws IOException, JSONException {
        InputStream is = new URL(url).openStream();
        try {
            BufferedReader br = new BufferedReader(new InputStreamReader(is, Charset.forName("UTF-8")));
            String jsonText = str(br);
            return new JSONObject(jsonText);
        }
        finally {
            is.close();
        }
    }
}

File: app/src/main/java/com/weiminal/biliwidget/DataToJson.kt
package com.weiminal.biliwidget

import kotlin.Throws
import org.json.JSONException
import org.json.JSONObject
import com.weiminal.biliwidget.DataToJson
import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader
import java.io.Reader
import java.lang.StringBuilder
import java.net.URL
import java.nio.charset.Charset

object DataToJson {
    @Throws(IOException::class)
    private fun str(r: Reader): String {
        val sb = StringBuilder()
        var a: Int
        while (r.read().also { a = it } != -1) {
            sb.append(a.toChar())
        }
        return sb.toString()
    }

    @Throws(IOException::class, JSONException::class)
    fun getData(url: String?): JSONObject {
        val `is` = URL(url).openStream()
        return try {
            val br = BufferedReader(
                InputStreamReader(
                    `is`,
                    Charset.forName("UTF-8")
                )
            )
            val jsonText = str(br)
            JSONObject(jsonText)
        } finally {
            `is`.close()
        }
    }
}